org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/editor/WSearchEditor.java

```diff
@@ -459,23 +459,23 @@ protected void actionText(String text)
 		}
 		if (log.isLoggable(Level.CONFIG)) log.config(getColumnName() + " - " + text);
 
-		int id = -1;
+		Object id = null;
 		
 		if (m_tableName == null)	//	sets table name & key column
 			setTableAndKeyColumn();
 		
 		final InfoPanel ip = InfoManager.create(lookup, gridField, m_tableName, m_keyColumnName, getComponent().getText(), false, getWhereClause());
 		if (ip != null && ip.loadedOK() && ip.getRowCount() == 1)
 		{
-			Integer key = ip.getFirstRowKey();
-			if (key != null && key.intValue() > 0)
+			Object key = ip.getFirstRowKey();
+			if (key != null && (key instanceof Integer && ((Integer)key).intValue() > 0 || key instanceof String && key.toString().length() > 0))
 			{
-				id = key.intValue();
+				id = key.toString();
```

Comment:
this could cause problem, when key is ID, the type of id have been changed from Integer to String.